cmd/query/app/flags.go

```diff
@@ -101,11 +111,13 @@ func (qOpts *QueryOptions) InitPortsConfigFromViper(v *viper.Viper, logger *zap.
 	qOpts.GRPCHostPort = v.GetString(queryGRPCHostPort)
 	qOpts.HostPort = ports.GetAddressFromCLIOptions(v.GetInt(queryPort), v.GetString(queryHostPort))
 
-	qOpts.TLS = tlsFlagsConfig.InitFromViper(v)
+	qOpts.TLSGRPC = tlsGRPCFlagsConfig.InitFromViper(v)
+	qOpts.TLSHTTP = tlsHTTPFlagsConfig.InitFromViper(v)
 
-	// query.host-port is not defined and at least one of query.grpc-server.host-port or query.http-server.host-port is defined.
+	// If either GRPC or HTTP servers use TLS, use dedicated ports.
+	// else-if query.host-port is not defined and at least one of query.grpc-server.host-port or query.http-server.host-port is defined.
 	// User intends to use separate GRPC and HTTP host:port flags
```

Comment:
These four lines of comment need to be rewritten, they are hard to grasp as they are right now. Also, the `if` statement below needs a refactoring: I'd rather have nested or seemingly duplicate code than have a complex statement. Like:

```go
if qOpts.TLSGRPC.Enabled || qOpts.TLSHTTP.Enabled {
    return qOpts
}

if !(v.IsSet(queryHostPort) || v.IsSet(queryPort)) && (v.IsSet(queryHTTPHostPort) || v.IsSet(queryGRPCHostPort)) {
    return qOpts
}
```